incubator/hnc/internal/validators/anchor_test.go

```diff
@@ -54,3 +49,53 @@ func TestSubnamespaces(t *testing.T) {
 		})
 	}
 }
+
+func TestAllowCascadingDeleteSubnamespaces(t *testing.T) {
+	// Create a chain of namespaces from "a" to "e", with "a" as the root. Among them,
+	// "b", "d" and "e" are subnamespaces. This is set up in a long chain to test that
+	// subnamespaces will look all the way up to get the 'allowCascadingDelete` value
+	// and won't stop looking when the first full namespace ancestor is met.
+	f := foresttest.Create("-AbCD")
+	h := &Anchor{Forest: f}
+
+	tests := []struct {
+		name string
+		op   v1beta1.Operation
+		acd  []string
```

Comment:
Since each namespace name is one character long, this could just be a single string